Library/Homebrew/diagnostic.rb

```diff
@@ -345,6 +338,40 @@ def check_access_prefix_directories
         EOS
       end
 
+      def check_missing_prefix_directories
+        return if HOMEBREW_PREFIX.to_s != "/usr/local"
+        missing_dirs = Keg::ALL_TOP_LEVEL_PATHS.reject(&:exist?)
+        return if missing_dirs.empty?
+
+        <<~EOS
+          The following directories are missing:
+          #{missing_dirs.join("\n")}
+          You should create them:
+            sudo mkdir -p #{missing_dirs.join(" ")}
+          And change the permissions of these directories:
+          #{permission_commands_for(missing_dirs).join("\n")}
+        EOS
+      end
+
+      def check_existing_prefix_directories_are_searchable
```

Comment:
I think think it's just worth checking if the desired directories are readable, writeable and searchable within a single function as they offer the same permission commands to fix them.